cdap-kafka/src/main/java/co/cask/cdap/kafka/run/KafkaServerMain.java

```diff
@@ -102,11 +94,25 @@ public void init(String[] args) {
       }
     }
 
-    int brokerId = generateBrokerId(address);
-    LOG.info(String.format("Initializing server with broker id %d", brokerId));
+    kafkaProperties = generateKafkaConfig(cConf);
+
+    Preconditions.checkState(Integer.parseInt(
+      kafkaProperties.getProperty(KafkaConstants.ConfigKeys.NUM_PARTITIONS_CONFIG)) > 0,
+                             "Num partitions should be greater than zero.");
+    int port = Integer.getInteger(kafkaProperties.getProperty(KafkaConstants.ConfigKeys.PORT_CONFIG), -1);
+    Preconditions.checkState(port > 0, "Port number is invalid.");
 
-    kafkaProperties = generateKafkaConfig(brokerId, zkConnectStr, hostname, port, numPartitions,
-                                          replicationFactor, logDir);
+    if (kafkaProperties.get("broker.id") == null) {
+      int brokerId = generateBrokerId(address);
+      LOG.info(String.format("Initializing server with broker id %d", brokerId));
+      kafkaProperties.setProperty("broker.id", Integer.toString(brokerId));
+    }
+    if (hostname == null) {
+      kafkaProperties.setProperty("host.name", null);
```

Comment:
yes, but what I'm trying to say is that it seems that the Kafka docs wants us to not set `host.name` to allow for the binding of all interfaces. Therefore, shouldn't we unset it if `HOSTNAME_CONFIG` is 0.0.0.0 instead of setting it to 0.0.0.0? I could also be not fully understanding what the code is doing.